### PR TITLE
Narrow enum_exists() arg to UnitEnum

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/10169

### CODE REVIEW

File: src/Type/Php/ClassExistsFunctionTypeSpecifyingExtension.php [hunk 1]
```diff
@@ -54,6 +55,17 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 			);
 		}
 
+		if ($functionReflection->getName() === 'enum_exists') {
+			return $this->typeSpecifier->create(
```
Comment (ondrejmirtes) [inline src/Type/Php/ClassExistsFunctionTypeSpecifyingExtension.php]: Please less code duplication here. The only thing different is the type argument.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
Thank you.

**Comment by ZejdCicak** [comment]
This makes `enum_exists` always treat an enum as UnitEnum, and never as BackedEnum, do you think there is a way to deal with this in library?

**Comment by ondrejmirtes** [comment]
@ZejdCicak Please open a new issue with a reproducing snippet of your problem.

**Pull request merged**